pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -143,6 +143,17 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 			validate: func(c client.Client, t *testing.T) {
 				provisions := getProvisions(c)
 				assert.Len(t, provisions, 1, "expected provision to exist")
+
+				pvc := &corev1.PersistentVolumeClaim{}
+				err := c.Get(context.TODO(), client.ObjectKey{Name: testInstallLogPVC().Name, Namespace: testNamespace}, pvc)
+				assert.NoError(t, err)
+				if pvc.Labels == nil {
+					t.Error("missing cluster deployment name label")
```

Comment:
This is already covered in the else case. You do not need any special code for when the labels are nil.

Reply:
Won't that just be a nil dereference error?

Maybe I misunderstood you, but I thought your feedback was "If the labels are empty, I think we would rather see asserts that the specific labels we are interesting in are missing rather than that the labels map is empty."

Do we no longer want to see "asserts that the specific labels we are interesting in are missing"?

Reply:
You will not have a nil dereference error. Getting an entry from a nil map returns the default value: It does not panic. See https://play.golang.org/p/iY5Wdz3BONO.

Reply:
Oh wow, that's interesting, I had no idea. Thanks, will fix!

Reply:
Sorry for the confusion. My comments could have been a lot clearer along the way.